src/main/java/com/example/homework10/Service/MusicServiceImpl.java

```diff
@@ -25,4 +26,12 @@ public Music createMusic(String title, String singer) {
         musicMapper.createMusic(MusicData);
         return MusicData;
     }
+
+
+    @Override
+    public Music updateMusic(int id, MusicUpdateForm musicUpdateForm) throws Exception {
+        Music updateMusic = musicUpdateForm.convertToMusic(id, musicUpdateForm);
+        musicMapper.updateMusic(updateMusic);
+        return updateMusic;
+    }
```

Comment:
完璧です！
最後にもう一点だけ、こちらの変換ですがControllerで行うようにしましょうか！
ServiceImplクラスとしては以下の形になります。
```suggestion
    public Music updateMusic(Music updateMusic) throws Exception {
        musicMapper.updateMusic(updateMusic);
        return updateMusic;
    }
```